auth.php

```diff
@@ -706,6 +721,29 @@ protected function handle_blocked_access() {
         }
     }
 
+    /**
+     * Checks configuration of the multiple IdP IP whitelist field. If the users IP matches, this will
+     * return the $md5idpentityid on true. Or false if not found.
+     *
+     * This is used in two places, firstly to determine if a saml redirect is to happen.
+     * Secondly to determine which IdP to force the redirect to.
+     *
+     * @return bool|string
+     */
+    protected function check_whitelisted_ip_redirect() {
+        $found = false;
+        foreach ($this->metadataentities as $idpentities) {
+            foreach ($idpentities as $md5idpentityid => $idpentity) {
+                $list = explode("\n", $idpentity->whitelist);
```

Comment:
look at remoteip_in_list() to remove a chunk of this code and also honor things like comments in the ip list

Reply:
I've made some changes for it to use address_in_subnet(), which takes a comma separated list of subnets. Grants greater flexibility than just a list of IPs